app/src/main/java/org/mozilla/focus/search/SearchEngineManager.java

```diff
@@ -241,6 +241,7 @@ private synchronized void loadFromDisk(Context context) {
         try {
             for (String engine : engines) {
                 final InputStream engineInputStream = new ByteArrayInputStream(prefs.getString(engine, "").getBytes(StandardCharsets.UTF_8));
+                // Search engine identifier is the search engine name.
```

Comment:
nit: what is the search engine identifier? `engine`? Maybe it should be renamed to `searchEngineName` or `searchEngineID` to make the comment clearer?

Maybe it'd be good to elaborate on how the code works. Right now I see:
- we get a value from the shared prefs about the engine
- This is passed into the engine object and becomes its identifier
- a name is extracted from the engine XML and that's the name field

wrt your comment, is the value in shared prefs the name such that `engine.identfier == engine.name`? If so, maybe the comment is too far extracted from the code where the names are set equal to each other that it'd go out of date... I'm not sure. Maybe there's a better way/place to specify this assumption?

Reply:
This is a throwback from integrating with the existing `SearchEngine` format for non-custom search engines. For that there's a separate engineName and engineIdentifier, but for custom engines, we just use engineName for both (because the user enters a name and a url). I'll add a comment into the SearchEngine class.